nativelib/src/main/resources/gc/commix/Marker.c

```diff
@@ -0,0 +1,380 @@
+                       Bytemap *bytemap, Object *object,
+                       ObjectMeta *objectMeta) {
+    assert(ObjectMeta_IsAllocated(objectMeta) ||
+           ObjectMeta_IsMarked(objectMeta));
+
+    assert(Object_Size(object) != 0);
+    Object_Mark(heap, object, objectMeta);
+
+    GreyPacket *out = *outHolder;
+    if (!GreyPacket_Push(out, object)) {
+        Marker_giveFullPacket(heap, stats, out);
+        *outHolder = out = Marker_takeEmptyPacket(heap, stats);
+        GreyPacket_Push(out, object);
+    }
+}
+
+void Marker_markConservative(Heap *heap, Stats *stats, GreyPacket **outHolder,
+                             word_t *address) {
+    assert(Heap_IsWordInHeap(heap, address));
+    Object *object = Object_GetUnmarkedObject(heap, address);
+    Bytemap *bytemap = heap->bytemap;
+    if (object != NULL) {
+        ObjectMeta *objectMeta = Bytemap_Get(bytemap, (word_t *)object);
+        assert(ObjectMeta_IsAllocated(objectMeta));
+        if (ObjectMeta_IsAllocated(objectMeta)) {
+            Marker_markObject(heap, stats, outHolder, bytemap, object,
+                              objectMeta);
+        }
+    }
+}
+
+int Marker_markRange(Heap *heap, Stats *stats, GreyPacket **outHolder,
+                     Bytemap *bytemap, word_t **fields, size_t length) {
+    int objectsTraced = 0;
+    word_t **limit = fields + length;
+    for (word_t **current = fields; current < limit; current++) {
+        word_t *field = *current;
+        if (Heap_IsWordInHeap(heap, field)) {
+            ObjectMeta *fieldMeta = Bytemap_Get(bytemap, field);
+            if (ObjectMeta_IsAllocated(fieldMeta)) {
+                Marker_markObject(heap, stats, outHolder, bytemap,
+                                  (Object *)field, fieldMeta);
+            }
+            objectsTraced += 1;
+        }
+    }
+    return objectsTraced;
+}
+
+int Marker_markRegularObject(Heap *heap, Stats *stats, Object *object,
+                             GreyPacket **outHolder, Bytemap *bytemap) {
+    int objectsTraced = 0;
+    int64_t *ptr_map = object->rtti->refMapStruct;
+    for (int64_t *current = ptr_map; *current != LAST_FIELD_OFFSET; current++) {
+        word_t *field = object->fields[*current];
+        if (Heap_IsWordInHeap(heap, field)) {
+            ObjectMeta *fieldMeta = Bytemap_Get(bytemap, field);
+            if (ObjectMeta_IsAllocated(fieldMeta)) {
+                Marker_markObject(heap, stats, outHolder, bytemap,
+                                  (Object *)field, fieldMeta);
+            }
+            objectsTraced += 1;
+        }
+    }
+    return objectsTraced;
+}
+
+int Marker_splitObjectArray(Heap *heap, Stats *stats, GreyPacket **outHolder,
+                            Bytemap *bytemap, word_t **fields, size_t length) {
+    word_t **limit = fields + length;
+    word_t **lastBatch =
+        fields + (length / ARRAY_SPLIT_BATCH) * ARRAY_SPLIT_BATCH;
+
+    assert(lastBatch <= limit);
+    // first
```

Comment:
This is an extremely uninformative comment. Either remove it or make it more clear.

Reply:
Removed